Subject: [PATCH] Re-add deleted_at to nested resource schemas

Speakeasy strips deleted_at from schemas that nest entity-root structs while retaining the field in the generated Go model. Restore it in all nine nested schemas so Terraform struct and object shapes remain identical.

diff --git a/internal/provider/app_entitlement_owner_entitlement_resource.go b/internal/provider/app_entitlement_owner_entitlement_resource.go
index 50efdb7c..4fab830b 100644
--- a/internal/provider/app_entitlement_owner_entitlement_resource.go
+++ b/internal/provider/app_entitlement_owner_entitlement_resource.go
@@ -108,6 +108,9 @@ func (r *AppEntitlementOwnerEntitlementResource) Schema(ctx context.Context, req
 						Computed:    true,
 						Description: `Flag to indicate if app-level access request defaults have been applied to the entitlement`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"deprovisioner_policy": schema.SingleNestedAttribute{
 						Computed: true,
 						Attributes: map[string]schema.Attribute{
diff --git a/internal/provider/app_entitlement_owner_user_resource.go b/internal/provider/app_entitlement_owner_user_resource.go
index 99debc81..8cb7bef5 100644
--- a/internal/provider/app_entitlement_owner_user_resource.go
+++ b/internal/provider/app_entitlement_owner_user_resource.go
@@ -90,6 +90,9 @@ func (r *AppEntitlementOwnerUserResource) Schema(ctx context.Context, req resour
 						Computed:    true,
 						Description: `The id of the user to whom tasks will be automatically reassigned to.`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"department": schema.StringAttribute{
 						Computed:    true,
 						Description: `The department which the user belongs to in the organization.`,
diff --git a/internal/provider/app_owner_entitlement_resource.go b/internal/provider/app_owner_entitlement_resource.go
index b84a7e18..1bf6cf8f 100644
--- a/internal/provider/app_owner_entitlement_resource.go
+++ b/internal/provider/app_owner_entitlement_resource.go
@@ -107,6 +107,9 @@ func (r *AppOwnerEntitlementResource) Schema(ctx context.Context, req resource.S
 						Computed:    true,
 						Description: `Flag to indicate if app-level access request defaults have been applied to the entitlement`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"deprovisioner_policy": schema.SingleNestedAttribute{
 						Computed: true,
 						Attributes: map[string]schema.Attribute{
diff --git a/internal/provider/app_owner_user_resource.go b/internal/provider/app_owner_user_resource.go
index 382b045d..9d681870 100644
--- a/internal/provider/app_owner_user_resource.go
+++ b/internal/provider/app_owner_user_resource.go
@@ -81,6 +81,9 @@ func (r *AppOwnerUserResource) Schema(ctx context.Context, req resource.SchemaRe
 						Computed:    true,
 						Description: `The id of the user to whom tasks will be automatically reassigned to.`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"department": schema.StringAttribute{
 						Computed:    true,
 						Description: `The department which the user belongs to in the organization.`,
diff --git a/internal/provider/app_resource_owner_entitlement_resource.go b/internal/provider/app_resource_owner_entitlement_resource.go
index e0d2a5c5..04d54f23 100644
--- a/internal/provider/app_resource_owner_entitlement_resource.go
+++ b/internal/provider/app_resource_owner_entitlement_resource.go
@@ -109,6 +109,9 @@ func (r *AppResourceOwnerEntitlementResource) Schema(ctx context.Context, req re
 						Computed:    true,
 						Description: `Flag to indicate if app-level access request defaults have been applied to the entitlement`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"deprovisioner_policy": schema.SingleNestedAttribute{
 						Computed: true,
 						Attributes: map[string]schema.Attribute{
diff --git a/internal/provider/app_resource_owner_user_resource.go b/internal/provider/app_resource_owner_user_resource.go
index 83bfbd38..f0bd30c1 100644
--- a/internal/provider/app_resource_owner_user_resource.go
+++ b/internal/provider/app_resource_owner_user_resource.go
@@ -99,6 +99,9 @@ func (r *AppResourceOwnerUserResource) Schema(ctx context.Context, req resource.
 						Computed:    true,
 						Description: `The id of the user to whom tasks will be automatically reassigned to.`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"department": schema.StringAttribute{
 						Computed:    true,
 						Description: `The department which the user belongs to in the organization.`,
diff --git a/internal/provider/connector_owner_entitlement_resource.go b/internal/provider/connector_owner_entitlement_resource.go
index 3f071aa7..0fa17a78 100644
--- a/internal/provider/connector_owner_entitlement_resource.go
+++ b/internal/provider/connector_owner_entitlement_resource.go
@@ -108,6 +108,9 @@ func (r *ConnectorOwnerEntitlementResource) Schema(ctx context.Context, req reso
 						Computed:    true,
 						Description: `Flag to indicate if app-level access request defaults have been applied to the entitlement`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"deprovisioner_policy": schema.SingleNestedAttribute{
 						Computed: true,
 						Attributes: map[string]schema.Attribute{
diff --git a/internal/provider/connector_owner_user_resource.go b/internal/provider/connector_owner_user_resource.go
index 113b4139..4b0dc85a 100644
--- a/internal/provider/connector_owner_user_resource.go
+++ b/internal/provider/connector_owner_user_resource.go
@@ -90,6 +90,9 @@ func (r *ConnectorOwnerUserResource) Schema(ctx context.Context, req resource.Sc
 						Computed:    true,
 						Description: `The id of the user to whom tasks will be automatically reassigned to.`,
 					},
+					"deleted_at": schema.StringAttribute{
+						Computed: true,
+					},
 					"department": schema.StringAttribute{
 						Computed:    true,
 						Description: `The department which the user belongs to in the organization.`,
diff --git a/internal/provider/directory_resource.go b/internal/provider/directory_resource.go
index e72dd052..5ee72f3e 100644
--- a/internal/provider/directory_resource.go
+++ b/internal/provider/directory_resource.go
@@ -95,6 +95,9 @@ func (r *DirectoryResource) Schema(ctx context.Context, req resource.SchemaReque
 							"created_at": schema.StringAttribute{
 								Computed: true,
 							},
+							"deleted_at": schema.StringAttribute{
+								Computed: true,
+							},
 							"merge_config": schema.SingleNestedAttribute{
 								Computed: true,
 								Attributes: map[string]schema.Attribute{
